---
title: "***tanggle***: an R package for visualization of phylogenetic networks in a *ggplot2* framework"
description: an R package for visualization of phylogenetic networks in a *ggplot2* framework
output: 
    html_vignette:
        toc: true
        toc_depth: 2
geometry: margin=2.5cm

vignette: >
  %\VignetteIndexEntry{***tanggle***: an R package for visualization of phylogenetic networks in a *ggplot2* framework}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
  \usepackage{lastpage}
  \usepackage{fancyhdr}
  \pagestyle{fancy}
  \fancyhead[CO,CE]{Your Document Header}
  \fancyfoot[CO,CE]{And this is a fancy footer}
  \fancyfoot[RE,RO]{\thepage}
bibliography: tanggle_references.bib
---
<!--- Define Headers and Footers --->
\fancypagestyle{plain}{%
  \renewcommand{\headrulewidth}{0pt}%
  \fancyhf{}%
  \fancyfoot[C]{\footnotesize Page \thepage\, of\, \pageref*{LastPage}}
  \setlength\footskip{0pt}
}
\pagestyle{plain}

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

\center
![](/Users/Marta/Documents/PROJECTS/SIDE_PROJECTS/SCHLIEP_SOLIS-LEMUS_tanggle/tanggle/inst/Logo.png){width=70%}
\center

Klaus Schliep^[klaus.schliep@gmail.com], Marta Vidal-García^[marta.vidalga@gmail.com], Leann Biancani,
Eren Ada, Francisco Henao Diaz, **XXX (anyone else I forgot)** & Claudia Solís-Lemus^[Claudia's email]

<br><br><br><br><br><br>

Last updated on `r Sys.Date()`.

\maketitle

\newpage

\tableofcontents

\newpage

\section{Introduction}
## 1. Introduction

***tanggle*** extends the *ggtree* R package [@Yu2017]  to allow
for the visualization of several types of phylogenetic networks using the
*ggplot2* [@Wickham2016] syntax. More specifically,
*tanggle* contains functions to allow the user to effectively plot: (1) split (i.e. implicit) networks
(unrooted, undirected) and (2) explicit networks (rooted, directed) with
reticulations. It offers an alternative to the plot functions already available
in *ape* [@Paradis2018] and *phangorn* [@Schliep2011].


XXX Add another paragraph or expand on the previous one

## 2.  List of functions

Function name | Brief description |
:-------- | :--------------------------------------------|
`geom_splitnet` | XXX description of this function and what it does
`ggsevonet` | XXX description of this function and what it does
`ggsplitnet` | XXX description of this function and what it does
`minimize_overlap` | XXX description of this function and what it does
`node.depth.evonet` | XXX description of this function and what it does
`reexports` | XXX description of this function and what it does


\section{Getting Started}
## 3. Getting started

Load the necessary libraries:

```{r, echo=TRUE, results='hide'}
packages_needed <- c("ggplot2", "ggtree", "phangorn")
packages_to_install <- packages_needed[!(packages_needed %in% installed.packages()[,"Package"])]
if(length(packages_to_install)) install.packages(packages_to_install)

```

If you need to install *ggtree* from github:
```{r, eval=FALSE}
library(devtools)
install_github("YuLab-SMU/ggtree")
```

Install *tanggle* from the github repository:
```{r, eval=FALSE}
library(devtools)
install_github("KlausVigo/tanggle")
```

And load all the libraries:
```{r, echo=TRUE, results='hide'}
library(devtools)
library(tanggle, quietly=T)
library(phangorn, quietly=T)
library(ggtree, quietly=T)
```

***

\section{Split Networks}
## 4. Split Networks

Split networks are data-display objects which allow for the definition of 2 (or
more) options for non-compatible splits. Split networks are most often used to
visualize consensus networks [@Holland2004] or neighbor-nets
[@Bryant2004]. This can be done either by using the `consensusNet` or
`neighbor-net` functions in *phangorn* [@Schliep2011] or by importing
nexus files from SplitsTree [@Huson2006].

\subsection{Data Types}
### 4.1. Data Types

*tanggle* accepts three forms of input data for split networks. The
following input options all generate a *networx* object for plotting.

* Nexus file created with SplitsTree [@Huson2006] and read with the
    `read.nexus.network` function in *phangorn* [@Schliep2011].
    
* Read in a split network in nexus format:
```{r}
fdir <- system.file("extdata/trees", package = "phangorn")
Nnet <- phangorn::read.nexus.networx(file.path(fdir,"woodmouse.nxs"))
```

2. A collection of gene trees (e.g.~from RAxML 
[@Stamatakis2014RAxML]) in one of the following formats:
    + Nexus file read with the function `read.nexus`
    + Text file in Newick format (one gene tree per line) read with the
        function `read.tree`
A consensus split network is then computed using the function `consensusNet`
in *phangorn* [@Schliep2011].
    
* Sequences in nexus, fasta or phylip format, read with the function
    `read.phyDat` in *phangorn* or the function $read.dna$ in
    *ape* [@Paradis2018]. Distances matrices are then computed for specific models of
    evolution using the function $dist.ml$ in *phangorn* [@Schliep2011] or $dist.dna$
    in *ape* [@Paradis2018]. From the distance matrix, a split network is reconstructed
    using the function $neighborNet$ in *phangorn* [@Schliep2011]. ***Optional***: branch
    lengths may be estimated using the function $splitsNetworks$ in
    *phangorn* [@Schliep2011].


\subsection{Plotting a Split Network}

### 4.2. Plotting a Split Network:
```{r}
ggsplitnet(Nnet) + geom_tiplab2()
```


Nodes can be annotated with $geom\_point$.
```{r}
ggsplitnet(Nnet) + geom_point(aes(shape=isTip, color=isTip), size=3)
```


\section{Plotting Explicit Networks}
### 4.3. Plotting Explicit Networks

The function $ggevonet$ plots explicit networks (phylogenetic trees
with reticulations). A recent addition to \CRANpkg{ape} \cite{Paradis2018} made
it possible to read in trees in extended newick format \cite{Cardona2008}.

<<>>=
## read in an explicit network (example from Fig. 2 in Cardona et al. 2008):
z <- read.evonet(text = "((1,((2,(3,(4)Y#H1)g)e,(((Y#H1,5)h,6)f)X#H2)c)a,
                 ((X#H2,7)d,8)b)r;")

#### plot an explicit network:
ggevonet(z) + geom_tiplab() + geom_nodelab()
p <- ggevonet(z, layout = "slanted") + geom_tiplab() + geom_nodelab()
p + geom_tiplab(size=3, color="purple")
p + geom_nodepoint(color="#b5e521", alpha=1/4, size=10)

@

\section{Summary}
## 5. Summary

The split network plots should take most of the functions compatible with
unrooted trees in ggtree. The explicit network plots are phylogram or slanted.
Not all options may work as intended yet.

\url{https://bioconductor.org/packages/devel/bioc/vignettes/ggtree/inst/doc/treeVisualization.html}



\nocite{Schliep2011}
\nocite{Schliep2017}
\nocite{Paradis2004}
% \nocite{Paradis2018}
\nocite{Holland2004}
\nocite{Yu2017}
\nocite{Bryant2004}
\nocite{Huson2006}
\nocite{Cardona2008}
\nocite{Wickham2016}


\newpage


\newpage
\appendix
\section{Session info}
## 6. Session info
```{r}
sessionInfo()
```

\bibliography{tanggle}
## 7. References


