<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>ggsplitnet — ggsplitnet • tanggle</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ggsplitnet — ggsplitnet"><meta name="description" content="drawing phylogenetic tree from phylo object"><meta property="og:description" content="drawing phylogenetic tree from phylo object"><meta property="og:image" content="https://klausvigo.github.io/tanggle/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">tanggle</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.15.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/tanggle_vignette_espanol.html">***tanggle***: Visualización de redes filogenéticas con *ggplot2*</a></li>
    <li><a class="dropdown-item" href="../articles/tanggle_vignette.html">***tanggle***: Visualization of phylogenetic networks in a *ggplot2* framework</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/KlausVigo/tanggle/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>ggsplitnet</h1>
      <small class="dont-index">Source: <a href="https://github.com/KlausVigo/tanggle/blob/master/R/ggsplitnet.R" class="external-link"><code>R/ggsplitnet.R</code></a></small>
      <div class="d-none name"><code>ggsplitnet.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>drawing phylogenetic tree from phylo object</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">ggsplitnet</span><span class="op">(</span><span class="va">tr</span>, mapping <span class="op">=</span> <span class="cn">NULL</span>, layout <span class="op">=</span> <span class="st">"slanted"</span>, mrsd <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  as.Date <span class="op">=</span> <span class="cn">FALSE</span>, yscale <span class="op">=</span> <span class="st">"none"</span>, yscale_mapping <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ladderize <span class="op">=</span> <span class="cn">FALSE</span>, right <span class="op">=</span> <span class="cn">FALSE</span>, branch.length <span class="op">=</span> <span class="st">"branch.length"</span>,</span>
<span>  ndigits <span class="op">=</span> <span class="cn">NULL</span>, angle <span class="op">=</span> <span class="fl">0</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-tr">tr<a class="anchor" aria-label="anchor" href="#arg-tr"></a></dt>
<dd><p>a networx object</p></dd>


<dt id="arg-mapping">mapping<a class="anchor" aria-label="anchor" href="#arg-mapping"></a></dt>
<dd><p>aes mapping</p></dd>


<dt id="arg-layout">layout<a class="anchor" aria-label="anchor" href="#arg-layout"></a></dt>
<dd><p>so far only 'slanted' is supported.</p></dd>


<dt id="arg-mrsd">mrsd<a class="anchor" aria-label="anchor" href="#arg-mrsd"></a></dt>
<dd><p>most recent sampling date</p></dd>


<dt id="arg-as-date">as.Date<a class="anchor" aria-label="anchor" href="#arg-as-date"></a></dt>
<dd><p>logical whether using Date class in time tree</p></dd>


<dt id="arg-yscale">yscale<a class="anchor" aria-label="anchor" href="#arg-yscale"></a></dt>
<dd><p>y scale</p></dd>


<dt id="arg-yscale-mapping">yscale_mapping<a class="anchor" aria-label="anchor" href="#arg-yscale-mapping"></a></dt>
<dd><p>yscale mapping for category variable</p></dd>


<dt id="arg-ladderize">ladderize<a class="anchor" aria-label="anchor" href="#arg-ladderize"></a></dt>
<dd><p>logical</p></dd>


<dt id="arg-right">right<a class="anchor" aria-label="anchor" href="#arg-right"></a></dt>
<dd><p>logical</p></dd>


<dt id="arg-branch-length">branch.length<a class="anchor" aria-label="anchor" href="#arg-branch-length"></a></dt>
<dd><p>variable for scaling branch, if 'none' draw cladogram</p></dd>


<dt id="arg-ndigits">ndigits<a class="anchor" aria-label="anchor" href="#arg-ndigits"></a></dt>
<dd><p>number of digits to round numerical annotation variable</p></dd>


<dt id="arg-angle">angle<a class="anchor" aria-label="anchor" href="#arg-angle"></a></dt>
<dd><p>rotate the plot.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>additional parameter</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>tree</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Schliep, K., Potts, A. J., Morrison, D. A. and Grimm, G. W.
(2017), Intertwining phylogenetic trees and networks.
<em>Methods Ecol Evol</em>. <b>8</b>, 1212–1220. doi:10.1111/2041-210X.12760</p>
<p>Dress, A.W.M. and Huson, D.H. (2004) Constructing Splits Graphs
<em>IEEE/ACM Transactions on Computational Biology and Bioinformatics
(TCBB)</em>, <b>1(3)</b>, 109–115</p>
<p>Bagci, C., Bryant, D., Cetinkaya, B. and Huson, D.H. (2021), Microbial
Phylogenetic Context Using Phylogenetic Outlines. <em>Genome Biology and
Evolution</em>. <b>13(9)</b>, evab213</p>
<p>Potts, A.J. and Hedderson, T.A. and Grimm, G.W. (2013), Constructing
Phylogenies in the Presence Of Intra-Individual Site Polymorphisms (2ISPs)
with a Focus on the Nuclear Ribosomal Cistron, <em>Systematic Biology</em>.
<b>63(1)</b>, 1–16</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html" class="external-link">ggtree</a></code>, <code><a href="https://klausvigo.github.io/phangorn/reference/as.networx.html" class="external-link">networx</a></code>,
<code><a href="https://klausvigo.github.io/phangorn/reference/consensusNet.html" class="external-link">consensusNet</a></code>, <code><a href="https://klausvigo.github.io/phangorn/reference/neighborNet.html" class="external-link">neighborNet</a></code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Klaus Schliep</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">yeast</span>, package<span class="op">=</span><span class="st">'phangorn'</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dm</span> <span class="op">&lt;-</span> <span class="fu">phangorn</span><span class="fu">::</span><span class="fu"><a href="https://klausvigo.github.io/phangorn/reference/dist.hamming.html" class="external-link">dist.ml</a></span><span class="op">(</span><span class="va">yeast</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">nnet</span> <span class="op">&lt;-</span> <span class="fu">phangorn</span><span class="fu">::</span><span class="fu"><a href="https://klausvigo.github.io/phangorn/reference/neighborNet.html" class="external-link">neighborNet</a></span><span class="op">(</span><span class="va">dm</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">ggsplitnet</span><span class="op">(</span><span class="va">nnet</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_tiplab2</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="ggsplitnet-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/KlausVigo/phangorn" class="external-link">phangorn</a></span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: ape</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘ape’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:ggtree’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rotate</span>
<span class="r-in"><span><span class="va">fdir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/examples"</span>, package <span class="op">=</span> <span class="st">"tanggle"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">nymania</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://klausvigo.github.io/phangorn/reference/read.phyDat.html" class="external-link">read.phyDat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">fdir</span>,</span></span>
<span class="r-in"><span>           <span class="st">"Nymania.capensis.ITS.alignment.fasta"</span><span class="op">)</span>, format<span class="op">=</span><span class="st">"fasta"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">nnet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://klausvigo.github.io/phangorn/reference/neighborNet.html" class="external-link">neighborNet</a></span><span class="op">(</span><span class="fu"><a href="https://klausvigo.github.io/phangorn/reference/dist.p.html" class="external-link">dist.p</a></span><span class="op">(</span><span class="va">nymania</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">ggsplitnet</span><span class="op">(</span><span class="va">nnet</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_tiplab2</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="ggsplitnet-2.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Klaus Schliep, Marta Vidal-Garcia, Claudia Solis-Lemus, Leann Biancani, Eren Ada, L. Francisco Henao Diaz, Joshua Justison.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

