<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>***tanggle***: Visualization of phylogenetic networks in a *ggplot2* framework (bioc) • tanggle</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="***tanggle***: Visualization of phylogenetic networks in a *ggplot2* framework (bioc)">
<meta property="og:description" content="tanggle">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">tanggle</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/tanggle_vignette.html">***tanggle***: an R package for visualization of phylogenetic networks in a *ggplot2* framework</a>
    </li>
    <li>
      <a href="../articles/tanggle_vignette_bioc.html">***tanggle***: Visualization of phylogenetic networks in a *ggplot2* framework (bioc)</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/KlausVigo/tanggle/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="tanggle_vignette_bioc_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>
<strong><em>tanggle</em></strong>: Visualization of phylogenetic networks in a <em>ggplot2</em> framework (bioc)</h1>
                        <h4 class="author">Klaus Schliep</h4>
            <address class="author_afil">
      Graz University of Technology<br><h4 class="author">Marta Vidal-García</h4>
            <address class="author_afil">
      University of Calgary<br><h4 class="author">Francisco Henao Diaz</h4>
                                    <h4 class="author">Eren Ada</h4>
                                    <h4 class="author">Claudia Solís-Lemus</h4>
            <address class="author_afil">
      University of Wisconsin-Madison<br><a class="author_email" href="mailto:#"></a><a href="mailto:klaus.schliep@gmail.com" class="email">klaus.schliep@gmail.com</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/KlausVigo/tanggle/blob/master/vignettes/tanggle_vignette_bioc.Rmd"><code>vignettes/tanggle_vignette_bioc.Rmd</code></a></small>
      <div class="hidden name"><code>tanggle_vignette_bioc.Rmd</code></div>

    </address>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>An R package for visualization of phylogenetic networks in a <em>ggplot2</em> framework</p>
    </div>
    

<p><img src="../inst/Logo.png" style="width:50.0%"></p>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Here we present a vignette for the R package <strong><em>tanggle</em></strong>, and provide an overview of its functions and their usage. <strong><em>Tanggle</em></strong> extends the <em>ggtree</em> R package <span class="citation">(Yu et al. 2017)</span> to allow for the visualization of several types of phylogenetic networks using the <em>ggplot2</em> <span class="citation">(Wickham 2016)</span> syntax. More specifically, <em>tanggle</em> contains functions to allow the user to effectively plot: (1) split (i.e. implicit) networks (unrooted, undirected) and (2) explicit networks (rooted, directed) with reticulations. It offers an alternative to the plot functions already available in <em>ape</em> <span class="citation">(Paradis and Schliep 2018)</span> and <em>phangorn</em> <span class="citation">(Schliep 2011)</span>.</p>
</div>
<div id="list-of-functions" class="section level1">
<h1 class="hasAnchor">
<a href="#list-of-functions" class="anchor"></a>List of functions</h1>
<table class="table">
<colgroup>
<col width="16%">
<col width="83%">
</colgroup>
<thead><tr class="header">
<th align="left">Function name</th>
<th align="left">Brief description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>geom_splitnet</code></td>
<td align="left">Adds a <em>splitnet</em> layer to a ggplot, to combine visualising data and the network</td>
</tr>
<tr class="even">
<td align="left"><code>ggsevonet</code></td>
<td align="left">Plots an explicit network from a <em>phylo</em> object</td>
</tr>
<tr class="odd">
<td align="left"><code>ggsplitnet</code></td>
<td align="left">Plots an implicit network from a <em>phylo</em> object</td>
</tr>
<tr class="even">
<td align="left"><code>minimize_overlap</code></td>
<td align="left">Reduces the number of reticulation lines crossing over in the plot</td>
</tr>
<tr class="odd">
<td align="left"><code>node.depth.evonet</code></td>
<td align="left">Returns the depths or heights of nodes and tips in the phylogenetic network</td>
</tr>
</tbody>
</table>
</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h1>
<p>Load the necessary libraries:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">packages_needed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ggplot2"</span>, <span class="st">"ggtree"</span>, <span class="st">"phangorn"</span><span class="op">)</span>
<span class="va">packages_to_install</span> <span class="op">&lt;-</span> <span class="va">packages_needed</span><span class="op">[</span><span class="op">!</span><span class="op">(</span><span class="va">packages_needed</span> <span class="op">%in%</span> 
                              <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span>,<span class="st">"Package"</span><span class="op">]</span><span class="op">)</span><span class="op">]</span>
<span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">packages_to_install</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="va">packages_to_install</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"ggtree"</span><span class="op">)</span></code></pre></div>
<p>If you need to install <em>ggtree</em> from github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://remotes.r-lib.org">remotes</a></span><span class="op">)</span>
<span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"YuLab-SMU/ggtree"</span><span class="op">)</span></code></pre></div>
<p>Install <em>tanggle</em> from the github repository:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://remotes.r-lib.org">remotes</a></span><span class="op">)</span>
<span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"KlausVigo/tanggle"</span><span class="op">)</span></code></pre></div>
<p>And load all the libraries:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://klausvigo.github.io/tanggle">tanggle</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/KlausVigo/phangorn">phangorn</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">ggtree</a></span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="split-networks" class="section level1">
<h1 class="hasAnchor">
<a href="#split-networks" class="anchor"></a>Split Networks</h1>
<p>Split networks are data-display objects which allow for the definition of 2 (or more) options for non-compatible splits. Split networks are most often used to visualize consensus networks <span class="citation">(Holland et al. 2004)</span> or neighbor-nets <span class="citation">(Bryant and Moulton 2004)</span>. This can be done either by using the <code>consensusNet</code> or <code>neighbor-net</code> functions in <em>phangorn</em> <span class="citation">(Schliep 2011)</span> or by importing nexus files from SplitsTree <span class="citation">(Huson and Bryant 2006)</span>.</p>
<div id="data-types" class="section level2">
<h2 class="hasAnchor">
<a href="#data-types" class="anchor"></a>Data Types</h2>
<p><em>tanggle</em> accepts three forms of input data for split networks. The following input options all generate a <em>networx</em> object for plotting.</p>
<ul>
<li><p>Nexus file created with SplitsTree <span class="citation">(Huson and Bryant 2006)</span> and read with the <code>read.nexus.network</code> function in <em>phangorn</em> <span class="citation">(Schliep 2011)</span>.</p></li>
<li><p>Read in a split network in nexus format:</p></li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fdir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/trees"</span>, package <span class="op">=</span> <span class="st">"phangorn"</span><span class="op">)</span>
<span class="va">Nnet</span> <span class="op">&lt;-</span> <span class="fu">phangorn</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phangorn/man/read.nexus.splits.html">read.nexus.networx</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">fdir</span>,<span class="st">"woodmouse.nxs"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>A collection of gene trees (e.g.~from RAxML <span class="citation">(Stamatakis 2014)</span>) in one of the following formats:
<ul>
<li>Nexus file read with the function <code>read.nexus</code>
</li>
<li>Text file in Newick format (one gene tree per line) read with the function <code>read.tree</code> A consensus split network is then computed using the function <code>consensusNet</code> in <em>phangorn</em> <span class="citation">(Schliep 2011)</span>.</li>
</ul>
</li>
</ol>
<ul>
<li>Sequences in nexus, fasta or phylip format, read with the function <code>read.phyDat</code> in <em>phangorn</em> <span class="citation">(Schliep 2011)</span> or the function <code>read.dna</code> in <em>ape</em> <span class="citation">(Paradis and Schliep 2018)</span>. Distances matrices are then computed for specific models of evolution using the function <code>dist.ml</code> in <em>phangorn</em> <span class="citation">(Schliep 2011)</span> or <code>dist.dna</code> in <em>ape</em> <span class="citation">(Paradis and Schliep 2018)</span>. From the distance matrix, a split network is reconstructed using the function <code>neighborNet</code> in <em>phangorn</em> <span class="citation">(Schliep 2011)</span>. <strong><em>Optional</em></strong>: branch lengths may be estimated using the function <code>splitsNetworks</code> in <em>phangorn</em> <span class="citation">(Schliep 2011)</span>.</li>
</ul>
</div>
<div id="plotting-a-split-network" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-a-split-network" class="anchor"></a>Plotting a Split Network:</h2>
<p>We can plot the network with the default options:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/ggsplitnet.html">ggsplitnet</a></span><span class="op">(</span><span class="va">Nnet</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab2.html">geom_tiplab2</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="tanggle_vignette_bioc_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>You can rename tip labels. Here we changed the names to species from 1 to 15:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Nnet</span><span class="op">$</span><span class="va">translate</span><span class="op">$</span><span class="va">label</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">Nnet</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>We can include the tip labels with <code>geom_tiplab2</code>, and customize some of the options. For example, here the tip labels are in blue and both in bold and italics, and we show the internal nodes in green:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/ggsplitnet.html">ggsplitnet</a></span><span class="op">(</span><span class="va">Nnet</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab2.html">geom_tiplab2</a></span><span class="op">(</span>col <span class="op">=</span> <span class="st">"blue"</span>, font <span class="op">=</span> <span class="fl">4</span>, hjust <span class="op">=</span> <span class="op">-</span><span class="fl">0.15</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodepoint.html">geom_nodepoint</a></span><span class="op">(</span>col <span class="op">=</span> <span class="st">"green"</span>, size <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></code></pre></div>
<p><img src="tanggle_vignette_bioc_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<p>Nodes can also be annotated with <code>geom_point</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/ggsplitnet.html">ggsplitnet</a></span><span class="op">(</span><span class="va">Nnet</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>shape<span class="op">=</span><span class="va">isTip</span>, color<span class="op">=</span><span class="va">isTip</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="tanggle_vignette_bioc_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
<div id="plotting-explicit-networks" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-explicit-networks" class="anchor"></a>Plotting Explicit Networks</h2>
<p>The function <code>ggevonet</code> plots explicit networks (phylogenetic trees with reticulations). A recent addition to <em>ape</em> <span class="citation">(Paradis and Schliep 2018)</span> made it possible to read in trees in extended newick format <span class="citation">(Cardona, Rosselló, and Valiente 2008)</span>.</p>
<p>Read in an explicit network (example from Fig. 2 in Cardona et al. 2008):</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">z</span> <span class="op">&lt;-</span> <span class="fu">read.evonet</span><span class="op">(</span>text <span class="op">=</span> <span class="st">"((1,((2,(3,(4)Y#H1)g)e,(((Y#H1,5)h,6)f)X#H2)c)a,
                 ((X#H2,7)d,8)b)r;"</span><span class="op">)</span></code></pre></div>
<p>Plot an explicit network:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/ggevonet.html">ggevonet</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodelab.html">geom_nodelab</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="tanggle_vignette_bioc_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ggevonet.html">ggevonet</a></span><span class="op">(</span><span class="va">z</span>, layout <span class="op">=</span> <span class="st">"slanted"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodelab.html">geom_nodelab</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span>, color<span class="op">=</span><span class="st">"purple"</span><span class="op">)</span></code></pre></div>
<p><img src="tanggle_vignette_bioc_files/figure-html/unnamed-chunk-12-2.png" width="700"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodepoint.html">geom_nodepoint</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"#b5e521"</span>, alpha<span class="op">=</span><span class="fl">1</span><span class="op">/</span><span class="fl">4</span>, size<span class="op">=</span><span class="fl">10</span><span class="op">)</span></code></pre></div>
<p><img src="tanggle_vignette_bioc_files/figure-html/unnamed-chunk-12-3.png" width="700"></p>
</div>
</div>
<div id="summary" class="section level1">
<h1 class="hasAnchor">
<a href="#summary" class="anchor"></a>Summary</h1>
<p>This vignette illustrates all the functions in the R package <strong><em>tanggle</em></strong>, and provides some examples on how to plot both explicit and implicit networks. The split network plots should take most of the functions compatible with unrooted trees in ggtree. The explicit network plots are phylogram or slanted.</p>
<p></p>
<p></p>
<div style="page-break-after: always;"></div>

</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.5 (2021-03-31)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.7</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] phangorn_2.6.3   ape_5.4-1        tanggle_0.99.0   ggtree_2.4.1    </span>
<span class="co">#&gt; [5] ggplot2_3.3.3    BiocStyle_2.18.1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_1.0.6          lattice_0.20-41     tidyr_1.1.3        </span>
<span class="co">#&gt;  [4] rprojroot_2.0.2     digest_0.6.27       utf8_1.2.1         </span>
<span class="co">#&gt;  [7] R6_2.5.0            evaluate_0.14       highr_0.8          </span>
<span class="co">#&gt; [10] pillar_1.6.0        rlang_0.4.10        lazyeval_0.2.2     </span>
<span class="co">#&gt; [13] Matrix_1.3-2        rmarkdown_2.7       pkgdown_1.6.1      </span>
<span class="co">#&gt; [16] textshaping_0.3.3   desc_1.3.0          labeling_0.4.2     </span>
<span class="co">#&gt; [19] stringr_1.4.0       igraph_1.2.6        munsell_0.5.0      </span>
<span class="co">#&gt; [22] compiler_4.0.5      xfun_0.22           pkgconfig_2.0.3    </span>
<span class="co">#&gt; [25] systemfonts_1.0.1   htmltools_0.5.1.1   tidyselect_1.1.0   </span>
<span class="co">#&gt; [28] tibble_3.1.0        bookdown_0.21       quadprog_1.5-8     </span>
<span class="co">#&gt; [31] codetools_0.2-18    fansi_0.4.2         crayon_1.4.1       </span>
<span class="co">#&gt; [34] dplyr_1.0.5         withr_2.4.1         grid_4.0.5         </span>
<span class="co">#&gt; [37] nlme_3.1-152        jsonlite_1.7.2      gtable_0.3.0       </span>
<span class="co">#&gt; [40] lifecycle_1.0.0     magrittr_2.0.1      scales_1.1.1       </span>
<span class="co">#&gt; [43] tidytree_0.3.3      stringi_1.5.3       cachem_1.0.4       </span>
<span class="co">#&gt; [46] farver_2.1.0        fs_1.5.0            ellipsis_0.3.1     </span>
<span class="co">#&gt; [49] ragg_1.1.2          rvcheck_0.1.8       generics_0.1.0     </span>
<span class="co">#&gt; [52] vctrs_0.3.7         fastmatch_1.1-0     tools_4.0.5        </span>
<span class="co">#&gt; [55] treeio_1.14.3       glue_1.4.2          purrr_0.3.4        </span>
<span class="co">#&gt; [58] parallel_4.0.5      fastmap_1.1.0       yaml_2.2.1         </span>
<span class="co">#&gt; [61] colorspace_2.0-0    BiocManager_1.30.12 aplot_0.0.6        </span>
<span class="co">#&gt; [64] memoise_2.0.0       knitr_1.32          patchwork_1.1.1</span></code></pre></div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>

<div id="refs" class="references">
<div id="ref-Bryant2004">
<p>Bryant, David, and Vincent Moulton. 2004. “Neighbor-Net: An Agglomerative Method for the Construction of Phylogenetic Networks.” <em>Molecular Biology and Evolution</em> 21 (2): 255–65. <a href="https://doi.org/10.1093/molbev/msh018">https://doi.org/10.1093/molbev/msh018</a>.</p>
</div>
<div id="ref-Cardona2008">
<p>Cardona, Gabriel, Francesc Rosselló, and Gabriel Valiente. 2008. “Extended Newick: It Is Time for a Standard Representation of Phylogenetic Networks.” <em>BMC Bioinformatics</em> 9 (1): 532. <a href="https://doi.org/10.1186/1471-2105-9-532">https://doi.org/10.1186/1471-2105-9-532</a>.</p>
</div>
<div id="ref-Holland2004">
<p>Holland, Barbara R., Katharina T. Huber, Vincent Moulton, and Peter J. Lockhart. 2004. “Using Consensus Networks to Visualize Contradictory Evidence for Species Phylogeny.” <em>Molecular Biology and Evolution</em> 21 (7): 1459–61. <a href="https://doi.org/10.1093/molbev/msh145">https://doi.org/10.1093/molbev/msh145</a>.</p>
</div>
<div id="ref-Huson2006">
<p>Huson, D. H., and D. Bryant. 2006. “Application of Phylogenetic Networks in Evolutionary Studies.” <em>Molecular Biology and Evolution</em> 23 (2): 254–67.</p>
</div>
<div id="ref-Paradis2018">
<p>Paradis, Emmanuel, and Klaus Schliep. 2018. “Ape 5.0: An Environment for Modern Phylogenetics and Evolutionary Analyses in R.” <em>Bioinformatics</em> 35 (3): 526–28.</p>
</div>
<div id="ref-Schliep2011">
<p>Schliep, Klaus Peter. 2011. “Phangorn: Phylogenetic Analysis in R.” <em>Bioinformatics</em> 27 (4): 592–93. <a href="https://doi.org/10.1093/bioinformatics/btq706">https://doi.org/10.1093/bioinformatics/btq706</a>.</p>
</div>
<div id="ref-Stamatakis2014RAxML">
<p>Stamatakis, A. 2014. “<span class="smallcaps">ra</span>X<span class="smallcaps">ml</span> Version 8: A Tool for Phylogenetic Analysis and Post-Analysis of Large Phylogenies.” <em>Bioinformatics</em> 30 (9): 1312–3.</p>
</div>
<div id="ref-Wickham2016">
<p>Wickham, Hadley. 2016. <em>Ggplot2: Elegant Graphics for Data Analysis</em>. Springer-Verlag New York. <a href="https://ggplot2.tidyverse.org">https://ggplot2.tidyverse.org</a>.</p>
</div>
<div id="ref-Yu2017">
<p>Yu, Guangchuang, David Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2017. “Ggtree: An R Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.” <em>Methods in Ecology and Evolution</em> 8 (1): 28–36. <a href="https://doi.org/10.1111/2041-210X.12628">https://doi.org/10.1111/2041-210X.12628</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Klaus Schliep, Marta Vidal-Garcia, Claudia Solis-Lemus, Leann Biancani, Eren Ada, L. Francisco Henao Diaz.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
