<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>***tanggle***: Visualización de redes filogenéticas con *ggplot2* • tanggle</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="***tanggle***: Visualización de redes filogenéticas con *ggplot2*">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">tanggle</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.15.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/tanggle_vignette_espanol.html">***tanggle***: Visualización de redes filogenéticas con *ggplot2*</a></li>
    <li><a class="dropdown-item" href="../articles/tanggle_vignette.html">***tanggle***: Visualization of phylogenetic networks in a *ggplot2* framework</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/KlausVigo/tanggle/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>***tanggle***: Visualización de redes filogenéticas con *ggplot2*</h1>
                        <h4 data-toc-skip class="author">Klaus
Schliep</h4>
            <address class="author_afil">
      Graz University of
Technology<br><h4 data-toc-skip class="author">Marta
Vidal-García</h4>
            <address class="author_afil">
      University of
Calgary<br><h4 data-toc-skip class="author">Leann
Biancani</h4>
            <address class="author_afil">
      University of Rhode
Island<br><h4 data-toc-skip class="author">Francisco
Henao-Diaz</h4>
            <address class="author_afil">
      University of British
Columbia<br><h4 data-toc-skip class="author">Eren
Ada</h4>
            <address class="author_afil">
      University of Rhode
Island<br><h4 data-toc-skip class="author">Claudia
Solís-Lemus</h4>
            <address class="author_afil">
      University of
Wisconsin-Madison<br><a class="author_email" href="mailto:#"></a><a href="mailto:klaus.schliep@gmail.com" class="email">klaus.schliep@gmail.com</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/KlausVigo/tanggle/blob/master/vignettes/tanggle_vignette_espanol.Rmd" class="external-link"><code>vignettes/tanggle_vignette_espanol.Rmd</code></a></small>
      <div class="d-none name"><code>tanggle_vignette_espanol.Rmd</code></div>
    </address>
</address>
</address>
</address>
</address>
</div>

    
    
<p><img src="../inst/extdata/logo.png" style="width:25.0%"></p>
<div class="section level2">
<h2 id="introducción">Introducción<a class="anchor" aria-label="anchor" href="#introducci%C3%B3n"></a>
</h2>
<p>Esta es la viñeta en español para el paquete de R
<strong>tanggle</strong>, en ella proveemos una vista general de sus
funciones y ejemplos de uso. <strong>Tanggle</strong> extiende el
paquete de R <strong>ggtree</strong> <span class="citation">(Yu et al.
2017)</span>, lo cual permite la visualización de múltiples tipos de
redes filogenéticas usando la sintaxis de <em>ggplot2</em> <span class="citation">(Wickham 2016)</span>. Especificamente,
<strong>tanggle</strong> contiene funciones que permiten al usuario
visualizar: (1) redes divididas o implícitas (no-enraizadas,
no-direccionadas) y (2) redes explícitas (enraizadas, direccionadas) con
reticulaciones. Estas funciones ofrecen alternativas a las funciones
gráficas disponibles en <em>ape</em> <span class="citation">(Paradis and
Schliep 2018)</span> y <em>phangorn</em> <span class="citation">(Schliep
2011)</span>.</p>
</div>
<div class="section level2">
<h2 id="lista-de-funciones">Lista de funciones<a class="anchor" aria-label="anchor" href="#lista-de-funciones"></a>
</h2>
<table class="table">
<colgroup>
<col width="16%">
<col width="83%">
</colgroup>
<thead><tr class="header">
<th align="left">name</th>
<th align="left">Brief description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>geom_splitnet</code></td>
<td align="left">Adds a <em>splitnet</em> layer to a ggplot, to combine
visualising data and the network</td>
</tr>
<tr class="even">
<td align="left"><code>ggevonet</code></td>
<td align="left">Grafica una red explícita de un objeto
<em>phylo</em>
</td>
</tr>
<tr class="odd">
<td align="left"><code>ggsplitnet</code></td>
<td align="left">Grafica una red implícita de un objeto
<em>phylo</em>
</td>
</tr>
<tr class="even">
<td align="left"><code>minimize_overlap</code></td>
<td align="left">Reduce el número de líneas de reticulación
entrecruzadas en la gráfica</td>
</tr>
<tr class="odd">
<td align="left"><code>node_depth_evonet</code></td>
<td align="left">Devuelve las profundidades o alturas de los nodos y
puntas en la red filogenética</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="para-empezar">Para empezar<a class="anchor" aria-label="anchor" href="#para-empezar"></a>
</h2>
<p>Instalar el paquete desde Bioconductor directamente:</p>
<p>Install the package from Bioconductor directly:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"tanggle"</span><span class="op">)</span></span></code></pre></div>
<p>O instalar la versión de desarrollo del paquete desde: <a href="https://github.com/KlausVigo/tanggle" class="external-link">Github</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"remotes"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"KlausVigo/tanggle"</span><span class="op">)</span></span></code></pre></div>
<p>Si necesita installer <em>ggtree</em> desde github:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"YuLab-SMU/ggtree"</span><span class="op">)</span></span></code></pre></div>
<p>Y cargar todas las librerías:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://klausvigo.github.io/tanggle">tanggle</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/KlausVigo/phangorn" class="external-link">phangorn</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://www.amazon.com/Integration-Manipulation-Visualization-Phylogenetic-Computational-ebook/dp/B0B5NLZR1Z/" class="external-link">ggtree</a></span><span class="op">)</span></span></code></pre></div>
<hr>
</div>
<div class="section level2">
<h2 id="redes-dividida-o-implicitas">Redes dividida o implicitas<a class="anchor" aria-label="anchor" href="#redes-dividida-o-implicitas"></a>
</h2>
<p>Las redes divididas son objetos de visualización de datos que
permiten la definición de 2 (o más) opciones de división no compatibles.
Las redes divididas son usadas frecuentemente para graficar redes
consenso <span class="citation">(Holland et al. 2004)</span> o redes
vecinas <span class="citation">(Bryant and Moulton 2004)</span>. Esto
puede llevarse a cabo utilizando las funciones <code>consensusNet</code>
o <code>neighbor-net</code> en <em>phangorn</em> <span class="citation">(Schliep 2011)</span>, o importando archivos Nexus
provenientes de SplitsTree <span class="citation">(Huson and Bryant
2006)</span>.</p>
</div>
<div class="section level2">
<h2 id="tipos-de-datos">Tipos de datos<a class="anchor" aria-label="anchor" href="#tipos-de-datos"></a>
</h2>
<p><em>tanggle</em> acepta tres formatos de entrada para redes
divididas. Las siguientes opciones de entrada generan un objeto
<em>network</em> para graficar.</p>
<ul>
<li><p>Archivo Nexus creado con SplitsTree <span class="citation">(Huson
and Bryant 2006)</span> e importado con la función
<code>read.nexus.network</code> en <em>phangorn</em> <span class="citation">(Schliep 2011)</span>.</p></li>
<li><p>Carga de red dividida en formato Nexus:</p></li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fdir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/trees"</span>, package <span class="op">=</span> <span class="st">"phangorn"</span><span class="op">)</span></span>
<span><span class="va">Nnet</span> <span class="op">&lt;-</span> <span class="fu">phangorn</span><span class="fu">::</span><span class="fu"><a href="https://klausvigo.github.io/phangorn/reference/read.nexus.splits.html" class="external-link">read.nexus.networx</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">fdir</span>,<span class="st">"woodmouse.nxs"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Una colección de árboles de genes (e.g., de RAxML <span class="citation">(Stamatakis 2014)</span>) en alguno de los siguientes
formatos: Importar archivo Nexus con la función <code>read.nexus</code>
Archivo de texto en formato Newick (un árbol de genes por línea)
importado con la función <code>read.tree</code>
</li>
</ol>
<p>Estimación de una red dividida consenso mediante la función
<code>consensusNet</code> en <em>phangorn</em> <span class="citation">(Schliep 2011)</span>.</p>
<ul>
<li>Secuencias en Nexus, Fasta o formato Phylip importandas con la
función <code>read.phyDat</code> en <em>phangorn</em> <span class="citation">(Schliep 2011)</span> o la función
<code>read.dna</code> en <em>ape</em> <span class="citation">(Paradis
and Schliep 2018)</span>. Luego se calculan las matrices de distancia
para los modelos de evolución específicos utilizando la función
<code>dist.ml</code> en <em>phangorn</em> <span class="citation">(Schliep 2011)</span> o <code>dist.dna</code> en
<em>ape</em> <span class="citation">(Paradis and Schliep 2018)</span>.
Con base en las matrices de distancia, se reconstruye una red dividida
utilizando la función <code>neighborNet</code> en <em>phangorn</em>
<span class="citation">(Schliep 2011)</span>.
<strong><em>Opcional</em></strong>: las longitudes de las ramas pueden
ser estimadas utilizando la función <code>splitsNetworks</code> en
<em>phangorn</em> <span class="citation">(Schliep 2011)</span>.</li>
</ul>
<div class="section level3">
<h3 id="para-graficar-una-red-dividida">Para graficar una Red Dividida<a class="anchor" aria-label="anchor" href="#para-graficar-una-red-dividida"></a>
</h3>
<p>Podemos graficar una red con las siguientes opciones por defecto:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ggsplitnet.html">ggsplitnet</a></span><span class="op">(</span><span class="va">Nnet</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab2.html" class="external-link">geom_tiplab2</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">p</span></span></code></pre></div>
<p><img src="tanggle_vignette_espanol_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
<p>Luego podemos establecer los límites para los ejes x &amp; y
permitiendo la lectura de los nombres de los ejes.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">xlim</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.019</span>, <span class="fl">.003</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">ylim</a></span><span class="op">(</span><span class="op">-</span><span class="fl">.01</span>, <span class="fl">.012</span><span class="op">)</span> </span>
<span><span class="va">p</span></span></code></pre></div>
<p><img src="tanggle_vignette_espanol_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<p>Es posible renombrar las puntas. Aquí cambiamos los nombres por un
consecutivo de 1 a 15:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Nnet</span><span class="op">$</span><span class="va">translate</span><span class="op">$</span><span class="va">label</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">Nnet</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span></span></code></pre></div>
<p>Podemos incluir los nombres de las puntas con
<code>geom_tiplab2</code>, y con esto personalizar algunas de sus
opciones. Por ejemplo, las puntas de color azul, en negrilla e itálicas;
también los nodos internos en verde:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggsplitnet.html">ggsplitnet</a></span><span class="op">(</span><span class="va">Nnet</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab2.html" class="external-link">geom_tiplab2</a></span><span class="op">(</span>col <span class="op">=</span> <span class="st">"blue"</span>, font <span class="op">=</span> <span class="fl">4</span>, hjust <span class="op">=</span> <span class="op">-</span><span class="fl">0.15</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodepoint.html" class="external-link">geom_nodepoint</a></span><span class="op">(</span>col <span class="op">=</span> <span class="st">"green"</span>, size <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></span></code></pre></div>
<p><img src="tanggle_vignette_espanol_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>Los nodos pueden ser anotados con <code>geom_point</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggsplitnet.html">ggsplitnet</a></span><span class="op">(</span><span class="va">Nnet</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>shape <span class="op">=</span> <span class="va">isTip</span>, color <span class="op">=</span> <span class="va">isTip</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="tanggle_vignette_espanol_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="para-graficar-una-red-explícita">Para graficar una Red Explícita<a class="anchor" aria-label="anchor" href="#para-graficar-una-red-expl%C3%ADcita"></a>
</h3>
<p>La función <code>ggevonet</code> dibuja redes explícitas (árboles
filogenéticos reticulados). Una adición reciente en <em>ape</em> <span class="citation">(Paradis and Schliep 2018)</span> permite importar
árboles en un formato Newick extendido <span class="citation">(Cardona,
Rosselló, and Valiente 2008)</span>.</p>
<p>Importar una red explícita (ejemplo de Fig. 2 en Cardona et
al. 2008):</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">z</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/evonet.html" class="external-link">read.evonet</a></span><span class="op">(</span>text <span class="op">=</span> <span class="st">"((1,((2,(3,(4)Y#H1)g)e,(((Y#H1,5)h,6)f)X#H2)c)a,</span></span>
<span><span class="st">                 ((X#H2,7)d,8)b)r;"</span><span class="op">)</span></span></code></pre></div>
<p>Para graficar una red explícita:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggevonet.html">ggevonet</a></span><span class="op">(</span><span class="va">z</span>, layout <span class="op">=</span> <span class="st">"rectangular"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html" class="external-link">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodelab.html" class="external-link">geom_nodelab</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="tanggle_vignette_espanol_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ggevonet.html">ggevonet</a></span><span class="op">(</span><span class="va">z</span>, layout <span class="op">=</span> <span class="st">"slanted"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html" class="external-link">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodelab.html" class="external-link">geom_nodelab</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html" class="external-link">geom_tiplab</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span>, color<span class="op">=</span><span class="st">"purple"</span><span class="op">)</span></span></code></pre></div>
<p><img src="tanggle_vignette_espanol_files/figure-html/unnamed-chunk-9-2.png" width="700"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodepoint.html" class="external-link">geom_nodepoint</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"#b5e521"</span>, alpha<span class="op">=</span><span class="fl">1</span><span class="op">/</span><span class="fl">4</span>, size<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p><img src="tanggle_vignette_espanol_files/figure-html/unnamed-chunk-9-3.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="resumen">Resumen<a class="anchor" aria-label="anchor" href="#resumen"></a>
</h2>
<p>Esta viñeta ilustra todas las funciones en el paquete
<strong><em>tanggle</em></strong> para R. Aquí se proveen algunos
ejemplos de como graficar redes implícitas y explícitas. La
visualización de redes divididas toma (se sirve de / utiliza ???) la
mayoría de las funciones compatibles con árboles no enraizados en
ggtree. Las opciones de diseño para las redes explícitas son
<em>rectangular</em> o <em>slanted</em>.</p>
<div style="page-break-after: always;"></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] phangorn_2.12.1  ape_5.8-1        tanggle_1.15.2   ggtree_3.16.2   </span></span>
<span><span class="co">#&gt; [5] ggplot2_3.5.2    BiocStyle_2.36.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] yulab.utils_0.2.0   sass_0.4.10         generics_0.1.4     </span></span>
<span><span class="co">#&gt;  [4] tidyr_1.3.1         ggplotify_0.1.2     lattice_0.22-7     </span></span>
<span><span class="co">#&gt;  [7] digest_0.6.37       magrittr_2.0.3      evaluate_1.0.4     </span></span>
<span><span class="co">#&gt; [10] grid_4.5.1          RColorBrewer_1.1-3  bookdown_0.43      </span></span>
<span><span class="co">#&gt; [13] fastmap_1.2.0       Matrix_1.7-3        jsonlite_2.0.0     </span></span>
<span><span class="co">#&gt; [16] BiocManager_1.30.26 purrr_1.0.4         aplot_0.2.8        </span></span>
<span><span class="co">#&gt; [19] scales_1.4.0        codetools_0.2-20    lazyeval_0.2.2     </span></span>
<span><span class="co">#&gt; [22] textshaping_1.0.1   jquerylib_0.1.4     cli_3.6.5          </span></span>
<span><span class="co">#&gt; [25] rlang_1.1.6         tidytree_0.4.6      withr_3.0.2        </span></span>
<span><span class="co">#&gt; [28] cachem_1.1.0        yaml_2.3.10         tools_4.5.1        </span></span>
<span><span class="co">#&gt; [31] parallel_4.5.1      dplyr_1.1.4         fastmatch_1.1-6    </span></span>
<span><span class="co">#&gt; [34] vctrs_0.6.5         R6_2.6.1            gridGraphics_0.5-1 </span></span>
<span><span class="co">#&gt; [37] lifecycle_1.0.4     fs_1.6.6            ggfun_0.1.9        </span></span>
<span><span class="co">#&gt; [40] treeio_1.32.0       ragg_1.4.0          pkgconfig_2.0.3    </span></span>
<span><span class="co">#&gt; [43] desc_1.4.3          pkgdown_2.1.3       pillar_1.11.0      </span></span>
<span><span class="co">#&gt; [46] bslib_0.9.0         gtable_0.3.6        glue_1.8.0         </span></span>
<span><span class="co">#&gt; [49] Rcpp_1.1.0          systemfonts_1.2.3   xfun_0.52          </span></span>
<span><span class="co">#&gt; [52] tibble_3.3.0        tidyselect_1.2.1    knitr_1.50         </span></span>
<span><span class="co">#&gt; [55] farver_2.1.2        igraph_2.1.4        patchwork_1.3.1    </span></span>
<span><span class="co">#&gt; [58] htmltools_0.5.8.1   nlme_3.1-168        labeling_0.4.3     </span></span>
<span><span class="co">#&gt; [61] rmarkdown_2.29      compiler_4.5.1      quadprog_1.5-8</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Bryant2004" class="csl-entry">
Bryant, David, and Vincent Moulton. 2004.
<span>“<span>Neighbor-Net</span>: An Agglomerative Method for the
Construction of Phylogenetic Networks.”</span> <em>Molecular Biology and
Evolution</em> 21 (2): 255–65. <a href="https://doi.org/10.1093/molbev/msh018" class="external-link">https://doi.org/10.1093/molbev/msh018</a>.
</div>
<div id="ref-Cardona2008" class="csl-entry">
Cardona, Gabriel, Francesc Rosselló, and Gabriel Valiente. 2008.
<span>“Extended Newick: It Is Time for a Standard Representation of
Phylogenetic Networks.”</span> <em>BMC Bioinformatics</em> 9 (1): 532.
<a href="https://doi.org/10.1186/1471-2105-9-532" class="external-link">https://doi.org/10.1186/1471-2105-9-532</a>.
</div>
<div id="ref-Holland2004" class="csl-entry">
Holland, Barbara R., Katharina T. Huber, Vincent Moulton, and Peter J.
Lockhart. 2004. <span>“Using Consensus Networks to Visualize
Contradictory Evidence for Species Phylogeny.”</span> <em>Molecular
Biology and Evolution</em> 21 (7): 1459–61. <a href="https://doi.org/10.1093/molbev/msh145" class="external-link">https://doi.org/10.1093/molbev/msh145</a>.
</div>
<div id="ref-Huson2006" class="csl-entry">
Huson, D. H., and D. Bryant. 2006. <span>“Application of Phylogenetic
Networks in Evolutionary Studies.”</span> <em>Molecular Biology and
Evolution</em> 23 (2): 254–67.
</div>
<div id="ref-Paradis2018" class="csl-entry">
Paradis, Emmanuel, and Klaus Schliep. 2018. <span>“Ape 5.0: An
Environment for Modern Phylogenetics and Evolutionary Analyses in
r.”</span> <em>Bioinformatics</em> 35 (3): 526–28.
</div>
<div id="ref-Schliep2011" class="csl-entry">
Schliep, Klaus Peter. 2011. <span>“Phangorn: Phylogenetic Analysis in
<span>R</span>.”</span> <em>Bioinformatics</em> 27 (4): 592–93. <a href="https://doi.org/10.1093/bioinformatics/btq706" class="external-link">https://doi.org/10.1093/bioinformatics/btq706</a>.
</div>
<div id="ref-Stamatakis2014RAxML" class="csl-entry">
Stamatakis, A. 2014. <span>“<span class="smallcaps">ra</span>x<span class="smallcaps">ml</span> Version 8: A Tool for Phylogenetic Analysis
and Post-Analysis of Large Phylogenies.”</span> <em>Bioinformatics</em>
30 (9): 1312–13.
</div>
<div id="ref-Wickham2016" class="csl-entry">
Wickham, Hadley. 2016. <em>Ggplot2: Elegant Graphics for Data
Analysis</em>. Springer-Verlag New York. <a href="https://ggplot2.tidyverse.org" class="external-link">https://ggplot2.tidyverse.org</a>.
</div>
<div id="ref-Yu2017" class="csl-entry">
Yu, Guangchuang, David Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk
Lam. 2017. <span>“Ggtree: An r Package for Visualization and Annotation
of Phylogenetic Trees with Their Covariates and Other Associated
Data.”</span> <em>Methods in Ecology and Evolution</em> 8: 28–36. <a href="https://doi.org/10.1111/2041-210X.12628" class="external-link">https://doi.org/10.1111/2041-210X.12628</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Klaus Schliep, Marta Vidal-Garcia, Claudia Solis-Lemus, Leann Biancani, Eren Ada, L. Francisco Henao Diaz, Joshua Justison.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
